src/util/ecdsa.rs

```diff
@@ -37,6 +37,23 @@ pub struct PublicKey {
 }
 
 impl PublicKey {
+    /// Constructs compressed ECDSA PublicKey from the provided generic Secp256k1 public key
+    pub fn new(pk: secp256k1::PublicKey) -> PublicKey {
+        PublicKey {
+            compressed: true,
+            key: pk,
```

Comment:
Minor nit, `pk` could be renamed to key which would allow to write `key,` instead of `key: pk,`.